gobblin-dataset-config/src/main/java/gobblin/dataset/config/ConfigClient.java

```diff
@@ -0,0 +1,37 @@
+package gobblin.dataset.config;
+
+import java.util.*;
+
+/*
+ 
+ config-stores{
```

Comment:
We should be using the Java ServiceLoader API (http://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html). The Hadoop FileSystem implementations use a similar pattern.